File: pe23.py
import time
start=time.time()
abundant=[]
for i in range(12, 21824):
	f=1
	for j in range(2, i//2+1):
		if(i%j==0):
			f+=j
			d=i//j
			if(j!=d):
				f+=d
			if(f>i):
				abundant.append(i)
				break
can=[i for i in range(21824)]
for i in range(len(abundant)):
	for j in range(i,len(abundant)):
		n=abundant[i]+abundant[j]
		if(n>=21824):
			break
		can[n]=0
print(sum(can))	

File: pe23.java
class pe23
{
	public static void main(String args[])
	{
		long i,f,j,d,k=0,abundant[]=new long[21812];
		for(i=12;i<21824;i++)
		{
			f=1;
			for(j=2;j<i/2;j++)
			{
				if(i%j==0)
				{
					f+=j;
					d=i/j;
					if(j!=d)
						f+=d;
					if(f>i)
					{
						abundant[(long)k++]=i;
						break;
					}
				}
			}
		}
	}
}